miden-crypto/src/merkle/index.rs

```diff
@@ -164,6 +164,18 @@ impl NodeIndex {
         self.depth = self.depth.saturating_sub(delta);
         self.value >>= delta as u32;
     }
+
+    // ITERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Return an iterator of the indices required for a Merkle proof of inclusion of a node at
+    /// `self`.
+    ///
+    /// This is *exclusive* on both ends: neither `self` nor the root index are included in the
+    /// returned iterator.
+    pub fn proof_indices(&self) -> impl ExactSizeIterator<Item = NodeIndex> + use<> {
```

Comment:
I haven't used this syntax yet. What does empty `use<>` do? Is it required?

Reply:
I seemed to remember, when the `-> impl Trait` syntax was being added, seeing a comment that ideally many of the iterator adapters in `std` should have used those instead of exposing the helper type. I can't seem to find this comment now though, so I actually have no idea what is considered better practice here.

Reply:
As for the empty `use<>`, that indicates that the returned type does not borrow from `&self`